utils/vlog/yosys/json.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+"""
+This is intended to provide a range of helper functions around the output of
+Yosys' `write_json`. Depending on the tasks, this may need to be flattened
+and/or in AIG format. In any case, at minimum `proc` and usually `prep` should
+be used before outputting the JSON.
+"""
+
+import os, sys
+import json
+
+class YosysModule:
+    def __init__(self, name, module_data):
+        self.name = name
+        self.data = module_data
+
+    # Return the name of the module
+    def get_name(self):
+        return self.name
+
+    # Return a list of ports of a module (default: top), as a 3-tuple (name, width, dir)
```

Comment:
This should be a docstring in a format supported by [sphinx](http://www.sphinx-doc.org/en/stable/ext/napoleon.html).

I like the NumPy formatting;
```python

def get_ports(self):
    """List of ports on a module.

    Returns
    -------
    name : str or None
    width : int
        The width in bits
    dir : str
        The direction, should be either `in` or `out`
    """
    pass
```